src/arrayops.jl

```diff
@@ -28,14 +28,14 @@ ContiguousSubArray{T,N,P,
 
 Array types with contiguous first dimension.
 """
-ContiguousArray{T,N} = Union{DenseArray{T,N}, ContiguousSubArray{T,N}}
+ContiguousArray{T,N} = Union{DenseArray{T,N}, ContiguousSubArray{T,N}, Base.ReinterpretArray{T,1,T2,A} where {A <: Union{DenseArray{T2,N}, ContiguousSubArray{T2,N}}} where {T2}}
```

Comment:
I think this line is too long.  I think 92 characters/line is the usual choice.

Also, `1` in `ReinterpretArray{T,1,T2,A}` does not look good to me.  I think this should be `N` so that all the types in the `Union` has the same dimension.

Since `ReinterpretArray` is defined as

> ```julia
> struct ReinterpretArray{T,N,S,A<:AbstractArray{S, N}} <: AbstractArray{T, N}
> ```
>
> https://github.com/JuliaLang/julia/blob/8157ac692813c36e41f834b60444e17219d30b0b/base/reinterpretarray.jl#L8


using different "`N`" for the first and second `N` might confuse the compiler.

The same comments apply for `FastContiguousArray`.